packages/ra-ui-materialui/src/field/ImageField.tsx

```diff
@@ -48,8 +49,10 @@ export const ImageField = <
             <Root className={className} {...sanitizeFieldRestProps(rest)}>
                 <ul className={ImageFieldClasses.list}>
                     {sourceValue.map((file, index) => {
-                        const fileTitleValue = get(file, title) || title;
-                        const srcValue = get(file, src) || title;
+                        const fileTitleValue = title
+                            ? get(file, title, title)
+                            : title;
+                        const srcValue = src ? get(file, src, src) : src;
```

Comment:
you've changed the meaning there, shouldn't this be 
```suggestion
                        const srcValue = src ? get(file, src, title) : title;
```